deltachat-ios/Helper/Utils.swift

```diff
@@ -82,16 +82,25 @@ struct Utils {
     }
 
     public static func share(url: String, parentViewController: UIViewController, sourceItem: UIBarButtonItem) {
-        if let url = URL(string: url) {
-            let activityVC = UIActivityViewController(activityItems: [url], applicationActivities: nil)
-            activityVC.popoverPresentationController?.barButtonItem = sourceItem
-            parentViewController.present(activityVC, animated: true, completion: nil)
-        }
+        guard let url = URL(string: url) else { return }
+
+        Utils.share(url: url, parentViewController: parentViewController, sourceItem: sourceItem)
     }
 
     public static func share(text: String, parentViewController: UIViewController, sourceItem: UIBarButtonItem) {
-        let activityVC = UIActivityViewController(activityItems: [text], applicationActivities: nil)
-        activityVC.popoverPresentationController?.barButtonItem = sourceItem // iPad crashes without a source
+        guard let data = text.data(using: .unicode) else { return }
```

Comment:
`.unicode` works in general, however, generates UTF-16/USC-2 instead of UTF-8. this can result in issues on the receiving side, as UTF-16/USC-2 is not as widely supported as UTF-8.

however, this is no longer a blocker, i'll have a rought look if there isn't just a BOM missing or so.

Reply:
indeed, when adding a BOM, UTF-8 works as expected.

background: BOM (Byte-Order-Mark) is not _needed_ for an UTF-8 file, however, many apps and libs need that at the beginning of a UTF-8 file to identify it as such (otherwise, a UTF-8 file content looks same to an ASCII/ANSI file ... and, without having metadata, can only be distinguished by some heuristics - which some libs do and others do not)

```suggestion
        guard var data = text.data(using: .utf8) else { return }
        let bom = Data([0xEF, 0xBB, 0xBF])
        data.insert(contentsOf: bom, at: 0)
```

with this change, UTF-8 works in "Notes" pad as well as in "WhatsApp"